metadata_store_scripts/V2025.07.22.0__add_generate_masking_parameters_procedure.sql

```diff
@@ -0,0 +1,391 @@
+        LEFT JOIN conditional_date_formats AS cdf
+            ON f.identified_column = cdf.identified_column
+        WHERE
+            f.[date_format] IS NOT NULL
+            OR cdf.conditional_date_format IS NOT NULL
+    ),
+
+    -- date_format_parameters - Create JSON mapping with fallback default
+    date_format_parameters AS (
+        SELECT
+            COALESCE(
+                '{' + STRING_AGG(
+                    '"' + LOWER(dfr.encoded_column_name) + '":"' + dfr.final_date_format + '"',
+                    ','
+                ) + '}',
+                '{}'
+            ) AS [DateFormatAssignments]
+        FROM date_format_resolution AS dfr
+        WHERE dfr.final_date_format IS NOT NULL
+    ),
+
+    -- type_casting_parameters - Create casting arrays with fallback defaults
+    type_casting_parameters AS (
+        SELECT
+            COALESCE(
+                '[' + STRING_AGG('"' + f.identified_column + '"', ',') + ']', '[""]'
+            ) AS [ColumnsToCastAsStrings],
+            COALESCE(
+                '[' + STRING_AGG(
+                    CASE WHEN f.adf_type = 'binary' THEN '"' + f.identified_column + '"' END, ','
+                ) + ']', '[""]'
+            ) AS [ColumnsToCastBackToBinary],
+            COALESCE(
+                '[' + STRING_AGG(
+                    CASE WHEN f.adf_type = 'boolean' THEN '"' + f.identified_column + '"' END, ','
+                ) + ']', '[""]'
+            ) AS [ColumnsToCastBackToBoolean],
+            COALESCE(
+                '[' + STRING_AGG(
+                    CASE WHEN f.adf_type = 'date' THEN '"' + f.identified_column + '"' END, ','
+                ) + ']', '[""]'
+            ) AS [ColumnsToCastBackToDate],
+            COALESCE(
+                '[' + STRING_AGG(
+                    CASE WHEN f.adf_type = 'double' THEN '"' + f.identified_column + '"' END, ','
+                ) + ']', '[""]'
+            ) AS [ColumnsToCastBackToDouble],
+            COALESCE(
+                '[' + STRING_AGG(
+                    CASE WHEN f.adf_type = 'float' THEN '"' + f.identified_column + '"' END, ','
+                ) + ']', '[""]'
+            ) AS [ColumnsToCastBackToFloat],
+            COALESCE(
+                '[' + STRING_AGG(
+                    CASE WHEN f.adf_type = 'integer' THEN '"' + f.identified_column + '"' END, ','
+                ) + ']', '[""]'
+            ) AS [ColumnsToCastBackToInteger],
+            COALESCE(
+                '[' + STRING_AGG(
+                    CASE WHEN f.adf_type = 'long' THEN '"' + f.identified_column + '"' END, ','
+                ) + ']', '[""]'
+            ) AS [ColumnsToCastBackToLong],
+            COALESCE(
+                '[' + STRING_AGG(
+                    CASE WHEN f.adf_type = 'timestamp' THEN '"' + f.identified_column + '"' END, ','
+                ) + ']', '[""]'
+            ) AS [ColumnsToCastBackToTimestamp]
+        FROM ruleset_computed AS f
+        WHERE f.treat_as_string = 1
```

Comment:
nitpick: since you have but one table here, you shouldn't need to reference `f` throughout